src/main/java/com/example/auctionapp/seeder/DatabaseSeeder.java

```diff
@@ -23,25 +25,35 @@
 
     private BaseRepository<Role> roleRepository;
     private UserRepository userRepository;
+    private BaseRepository<Category> categoryRepository;
+    private BaseRepository<Subcategory> subcategoryRepository;
 
     private Logger logger = LoggerFactory.getLogger(RoleService.class);
```

Comment:
I guess i missed this.
This property should be `static`. 
`LoggerFactory.getLogger(RoleService.class);` is a line of code that does not have outside dependencies. It is purely instanced inside the DatabaseSeeder class. 
Furthermore, logger is a class that can be viewed as a singleton - there can be multiple loggers for a class, but, performance vise, it is best for there to be 1 Logger class for all (Exp. DatabaseSeeder) classes.
For example, DataSeeder1 instance and DataSeeder2 instance can both invoke `logger.log("test")` on a single logging class.

This is achieved by making this property `static` - making it the same for every instance of a `DatabaseSeeder` class.

Reply:
One more thing, this logger is created for `RoleService.class` but it should have been made for `DatabaseSeeder.class`